packages/legend-application-studio/src/stores/EditorStore.ts

```diff
@@ -1187,11 +894,73 @@ export class EditorStore implements CommandRegistrar {
     );
   }
 
-  changeLocalChangesState(): void {
-    if (this.isInFormMode) {
-      this.localChangesState = new FormLocalChangesState(this, this.sdlcState);
-    } else {
-      this.localChangesState = new TextLocalChangesState(this, this.sdlcState);
+  *switchModes(
+    to: GRAPH_EDITOR_MODE,
+    fallbackOptions?: {
```

Comment:
I guess we can sleep on this for now, worst case, we have different set of methods: `handleGraphBuilderFailure()` and `handleCompilationFailure()`